src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala

```diff
@@ -315,17 +333,128 @@ object AOIMonitorUtils extends Logging{
+      }
+    } finally {
+      if (connection != null) connection.close()
+    }
+  }
+
+  def queryAreasOfInterest(): List[AOIData] = {
+    var connection: Connection = null
+    val wkbReader = new WKBReader(new GeometryFactory(new PrecisionModel(), 4326))
+    val data = ListBuffer[AOIData]()
+    try {
+      connection = AOIDatabaseConfig.getConnection
+      val query =
+        """
+          |SELECT
+          |    aoi.id as aoi_id,
+          |    user_id,
+          |    aoi.created_at as created_at,
+          |    ST_AsBinary(geom) as geometry,
+          |    name,
+          |    email
+          |FROM area_of_interest aoi
+          |LEFT JOIN user_info u
+          |ON aoi.user_id = u.id
+          |""".stripMargin
+      val preppedStatement =
+        connection.prepareStatement(query)
+      val rs = preppedStatement.executeQuery()
+      while (rs.next()) {
+        val aoiId = UUID.fromString(rs.getString("aoi_id"))
+        val userId = UUID.fromString(rs.getString("user_id"))
+        val createdAt = rs.getTimestamp("created_at")
+        val geom = wkbReader.read(rs.getBytes("geometry"))
+        val name = rs.getString("name")
+        val email = rs.getString("email")
+        val aoiData = AOIData(aoiId, userId, createdAt, geom, name, email)
+        data.prepend(aoiData)
+      }
+    } finally {
+      if (connection != null) connection.close()
+    }
+    data.toList
   }
+}
 
-  def getCurrentSequence(augmentedDiffSource: URI, procName: String)(implicit spark: SparkSession): Option[Int] = {
-    spark.sparkContext.parallelize(Seq(augmentedDiffSource)).map { uri =>
-      queryLastSequence(uri, procName)
-    }.collect.apply(0)
+case class AOIData(aoiId: UUID,
+                   userId: UUID,
+                   createdAt: Timestamp,
+                   geom: Geometry,
+                   name: String,
+                   email: String)
+
+sealed trait Interval {
+  def value: String
+  def shortCode: String
+}
+object Interval {
+  case object Weekly extends Interval {
+    val value = "weekly"
+    val shortCode = "w"
+  }
+  case object Daily extends Interval {
+    val value = "daily"
+    val shortCode = "d"
   }
 
-  def warnMessage: (=> String) => Unit = logWarning
+  val options: Set[Interval] = Set(Daily, Weekly)
+
+  def unapply(str: String): Option[Interval] = options.find(_.value == str)
+
+  implicit val intervalArgument: Argument[Interval] = new Argument[Interval] {
+    def read(str: String) = {
+      options.find(_.shortCode == str) match {
+        case Some(interval) => Validated.valid(interval)
+        case None           => Validated.invalidNel(s"Invalid interval: $str")
+      }
+    }
+
+    override def defaultMetavar: String = {
+      val opts = options
+        .map { i =>
+          Seq(i.shortCode, i.value).mkString("=")
+        }
+        .mkString(", ")
+      s"Period of time to aggregate over ($opts)"
+    }
+  }
+}
+
+object AOIDatabaseConfig {
```

Comment:
Removed the `--database-url` CLI argument in favor of this. See commit log for reasoning.